src/v/pandaproxy/rest/proxy.cc

```diff
@@ -41,9 +42,20 @@ auto wrap(Handler h) {
             // Will throw 400 & 401 if auth fails
             auto auth_result = rq.service().authenticator().authenticate(
               *rq.req);
-            // Will throw 403 if user enabled HTTP Basic Auth but
-            // did not give the authorization header.
-            auth_result.require_authenticated();
+            try {
+                auth_result.require_authenticated();
+            } catch (const ss::httpd::base_exception& ex) {
+                // The failure from require_authenticated() will throw 401
+                // instead of 403. So convert the exception to 403.
+                if (
+                  ex.status() == ss::httpd::reply::status_type::unauthorized) {
+                    throw ss::httpd::base_exception(
+                      ex.what(), ss::httpd::reply::status_type::forbidden);
+                } else {
+                    // Something else went wrong
+                    throw;
+                }
```

Comment:
can you add to the commit message why 403 instead of 401 is needed?

right above this block there is an existing authenticate call with the comment

```
            // Will throw 400 & 401 if auth fails
            auto auth_result = rq.service().authenticator().authenticate(
              *rq.req);
```

so it's not clear why then later a 401 gets converted into a 403.